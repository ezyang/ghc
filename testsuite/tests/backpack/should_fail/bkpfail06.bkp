unit p where
    signature H where
        data T = T Int
unit q where
    module T where
        data T = T Bool
    module H(T(..)) where
        import T
    include p
