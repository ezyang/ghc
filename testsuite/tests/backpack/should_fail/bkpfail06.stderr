[1 of 3] Processing p
  [1 of 2] Compiling H[sig]           ( p/H.hsig, nothing )
  [2 of 2] Compiling A                ( p/A.hs, nothing )
[2 of 3] Processing qimpl
  Instantiating main-qimpl
  [1 of 2] Compiling T                ( qimpl/T.hs, bkpfail06-out/main-qimpl/T.o )
  [2 of 2] Compiling H                ( qimpl/H.hs, bkpfail06-out/main-qimpl/H.o )
[3 of 3] Processing q
  Instantiating main-q
  [1 of 2] Including main-qimpl
  [2 of 2] Including main-p
    Instantiating main-p(main-qimpl:H)
    [1 of 2] Compiling H[sig]           ( p/H.hsig, nothing )

bkpfail06.bkp:3:9: error:
    Type constructor ‘T’ has conflicting definitions in the module
    and its hsig file
    Main module: data T = T Bool
    Hsig file:  data T = T Int
    The constructors do not match: The types for ‘T’ differ
